guides/giving_donations_integration.md

```diff
@@ -198,3 +214,38 @@ In this case, `Person` 14994497 (`person_to_remove`) was merged into 7504309 (`p
+
+- Someone makes a new donation
+- Giving dispatches a webhook event containing a data payload for this donation
+- Your integration's app/server receives the webhook (at a URL you've specified), and takes some appropriate action (perhaps storing this new donation in your database)
+
+## Setting up Webhooks
+
+To start using webhooks, you'll want to head over to the [Webhooks Dashboard](https://api.planningcenteronline.com/webhooks#/). From there you can add a new Subscription URL and select which events to subscribe to. That's it! You can have one Subscription URL that handles many events, or use separate Subscription URLs for each event, it's entirely up to you.
+
+**Note:** Since webhooks are created under your specific user account, they only return data you have permission to see within Planning Center. For Giving, that means that only users with the role of **Admin** and **Bookkeeper** are able to fully utilize all webhooks.
+
+## Best Practices
+
+When building out your integration's webhook handlers, there are a few things you'll want to keep in mind.
+
+### Duplicate Events
+
+It's possible you'll receive the same webhook event more than once. This is especially true of `updated` events, but can happen for `created` and `destroyed` as well. We recommend making your event processing [idempotent](https://en.wikipedia.org/wiki/Idempotence) to ensure that you can handle duplicate events. For example, you could log the events you've processed and skip the processing of an already-logged event.
+
+### Event Order
+
+We can't guarantee that you'll receive webhook events in the order they occurred, and you'll want to account for this in your event processing. As an example, it's possible you could receive a `donation.created` event before the corresponding `profile.created` event for a first time donor. In these cases, we recommend using our API to fetch any missing data you need (ie. making a request to the `/people/:id` endpoint to get details on the person who just donated).
+
+### First Time Donor Webhooks
+
+Speaking of first time donors, we realize this is an event you may be particularly interested in. Perhaps your system will want to send a special thank you email or record the date a person became a donor. It's important to note that the `profile.created` event does not necessarily indicate a first time donation, just that a new Giving profile was made (which can happen for a variety of reasons). Instead, you can use `profile.first_donated`, which we've added specifically for this purpose. You'll receive this webhook any time a donation is processed for a person with no existing donations\*. Because Giving admins can reassign or delete donations, it is possible you will receive this webhook more than once for the same person.
```

Comment:
Wondering about this section and spitballing some other phrasing: 

Many integrations will want to take some action when a someone gives their first donation, either to take some special action (like sending a thank you email) or recording the date a person became a donor. 

For these purposes, the `profile.created` webhook might feel like the obvious choice. However, it's important to note that in Planning Center Giving, donor profiles can be created for all kinds of reasons – not just when someone donates. In fact, it's very possible for people to have a donor profile with a completely empty donation history. This webhook just tells you that a profile was created.

To find out when a person _actually_ gave their first donation, you'll want to use the `profile.first_donated` webhook which will return the associated donation ID that qualifies as this donor's first donation. *

\*. It's worth noting that since Giving admins can reassign or delete donations, this value could change. That means it's possible you will receive this webhook more than once for the same person if their underlying donation history changes.